pkg/controller/migration/convert/core.go

```diff
@@ -255,19 +257,36 @@ func handleAnnotations(c *components, _ *operatorv1.Installation) error {
 
 // removeExpectedAnnotations returns the given annotations with common k8s-native annotations removed.
 // this function also accepts a second argument of additional annotations to remove.
-func removeExpectedAnnotations(existing, ignore map[string]string) map[string]string {
+func removeExpectedAnnotations(existing, ignoreWithValue map[string]string, ignoreNoMatterWhatValue []string) map[string]string {
 	a := existing
+	annotationKeyRegexps := make([]*regexp.Regexp, 0)
+	for _, annotationKey := range ignoreNoMatterWhatValue {
+		annotationKeyRegexp, err := regexp.Compile(annotationKey)
+		if err != nil {
+			log.Error(fmt.Errorf("%s is not a valid regular expression", annotationKey), "Error when removing expected annotations")
+			continue
+		}
+		annotationKeyRegexps = append(annotationKeyRegexps, annotationKeyRegexp)
+	}
```

Comment:
Minor: Seems like `removeExpectedAnnotations(...)` is called in various places (i.e. multiple times). It would be slightly more efficient to build these reg exps (`regexp.Compile(...)`) outside of this function, so they only need to be compiled once. Then they just need to be passed around. 

Going all the way out, perhaps instead of doing them within the scope of these functions, we could have an `init(...)` in this package that loads the regexes into a package level variable.